### PR TITLE
Initial first stab at the documentation

### PR DESCRIPTION
**Pull request by stjohnjohnson**
Just starting the foundation for it.

### CODE REVIEW

File: docs/reference/environment.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Environment Variables
+
+Screwdriver exports a set of environment variables that you can rely on during
+the course of a build. These are:
+
+## Continuous Integration
+| Name | Description |
+|------|-------------|
+| SCREWDRIVER | Set to true if we are running in Screwdriver |
+| CI | Set to true if we are running in Screwdriver |
```
Comment (jpcollins) [inline docs/reference/environment.md]: I assume these CI related variables are legacy?  Are we carrying these forward?

Comment (stjohnjohnson) [inline docs/reference/environment.md]: This is actually used in Travis and Wercker https://docs.travis-ci.com/user/environment-variables/#Default-Environment-Variables

File: README.md [hunk 2]
```diff
@@ -1,2 +1,49 @@
-# guide
-Screwdriver Documentation
+# Screwdriver Guide
+
+## Installation of MkDocs
+In order to install MkDocs you'll need [Python] installed on your system, as
```
Comment (d2lam) [inline README.md]: Is `[Python]` supposed to be a link?

File: docs/getting-started/domain.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# Domain Model
+
+## Step
+
+A step is a named action that needs to be performed, usually a single shell command.  If the command finishes with a non-zero exit code, the step is considered a failure.
+
+## Metadata
+
+Metadata is a structured key/value storage of relevant information about an instance of a Job.  This is automatically populated with basic information like Git SHA1 and start/stop time.  It can be updated throughout the job.
```
Comment (d2lam) [inline docs/getting-started/domain.md]: why is 'Job' capitalized here?

File: docs/getting-started/domain.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+Jobs share four pieces of context between steps:
+
+ - Source Code - Checked out to a given revision at the start of the job
+ - Filesystem - Any changes made by a step to the filesystem will be visible to subsequent steps
+ - Container - Any processes started or stopped by a step will be visible to subsequent steps
+ - Metadata - Structured key/value storage of relevant information
+
+Jobs are started by changes in source code, as a trigger from an upstream job, or by human decision.
+
```
Comment (d2lam) [inline docs/getting-started/domain.md]: Do we want to add Build as well?

File: docs/index.md [hunk 5]
```diff
@@ -0,0 +1,10 @@
+
+# screw·driv·er
+
+> noun
+
+> 1. a tool for turning screws so as to drive them into their place.
+
+> 2. vodka and orange juice served with ice.
+
+> 3. a service for easily building and deploying software.
```
Comment (tkyi) [inline docs/index.md:10]: you're missing a 4th definition

`4. a software engineer at Yahoo, St.John Johnson`

File: docs/reference/configuration.md [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+ - [Wercker](http://devcenter.wercker.com/docs/wercker-yml/index.html)
+ - [Travis](https://docs.travis-ci.com/user/getting-started/)
+ - [Drone.io](http://readme.drone.io/usage/overview/)
+
+## Format
+
+We chose to use standard YAML formatting as it provided:
+
+ - a clean interface
+ - human readable
```
Comment (d2lam) [inline docs/reference/configuration.md]: Make it a noun: `human readability`  to keep it consistent

File: docs/reference/configuration.md [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+
+## Global Features
+
+These are specified at the root of the `screwdriver.yaml` and apply to the entire pipeline.
+
+### Workflow
+
+> I want to easily describe the flow of my pipeline, so that it is clear what jobs will run in what order.
+
+In the past we had users specifying `triggers` in each job - which defined what jobs were started on the completion of that job.  This was disjointed and did not provide an easy overview of the flow of the pipeline.  Because of that, we added a basic `order` feature - an array of jobs to execute in that order.  This worked for most simple pipelines, but was unusable when users wanted to do parallelization or detached jobs.  Workflow is the next iteration of this, a clean representation of the flow of a pipeline.
```
Comment (d2lam) [inline docs/reference/configuration.md]: Do we want to include all this? I'm reading this from a **user's point of view** and this seems like a lot of unnecessary information. I don't care too much about how screwdriver used to be, or why the team decided to make certain decisions. I only care about how screwdriver works **now** and how to use it, especially since this is the configuration page. Maybe it's more appropriate to include our design decision on a separate page?

File: docs/getting-started/domain.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+A job consists of multiple steps, each of which will be run in series.  If any step fails, then the entire job is considered failed and subsequent steps will be skipped (unless configured otherwise).
+
+Jobs share four pieces of context between steps:
+
+ - Source Code - Checked out to a given revision at the start of the job
+ - Filesystem - Any changes made by a step to the filesystem will be visible to subsequent steps
+ - Container - Any processes started or stopped by a step will be visible to subsequent steps
+ - Metadata - Structured key/value storage of relevant information
+
+Jobs are started by changes in source code, as a trigger from an upstream job, or by human decision.
```
Comment (tkyi) [inline docs/getting-started/domain.md]: might want to reword this to be more clear; something like this:

> Various events can trigger jobs to start, such as changing the source code, completing an upstream job, or manually clicking `Start` in the UI.

File: docs/index.md (hunk 5 above)
Comment (d2lam) [inline docs/index.md:10]: ^^^ 👍  Please add 😄

File: docs/getting-started/domain.md [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+
+## Pipeline
+
+A pipeline represents a collection of jobs that share the same source code. Two jobs are required to be defined in every pipeline:
+
+ - Pull-Request - Dynamic job that is created on each proposed change to the source code (via GitHub Pull Requests)
+ - Main - Job that runs on each change to the source code.
```
Comment (tkyi) [inline docs/getting-started/domain.md]: Would renaming the default job to `Build` make more sense?

File: docs/getting-started/domain.md [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+
+## Platform
+
+A platform is a combination of pre-defined jobs, pre-defined steps, and pre-defined containers that make building and testing a type of artifact easy.  They are maintained by topic experts.
+
+## Container
+
+_@todo Describe containers_
```
Comment (tkyi) [inline docs/getting-started/domain.md]: Docker containers?

File: docs/reference/configuration.md (hunk 7 above)
Comment (tkyi) [inline docs/reference/configuration.md]: I agree with @d2lam , this seems a little wordy for a configuration page. It would be good to have on an architecture or User Guide sort of page (e.g., http://kubernetes.io/docs/user-guide/)

File: docs/reference/configuration.md [hunk 11]
```diff
@@ -0,0 +1,195 @@
...
+image: node
+# docker.example.com/_/ruby:2
+image: docker.example.com/_/ruby:2
+```
+
+### Matrix support
+
+> I want to test my source code against multiple versions of languages
+
+Screwdriver used to
```
Comment (tkyi) [inline docs/reference/configuration.md]: ???

File: docs/reference/configuration.md [hunk 12]
```diff
@@ -0,0 +1,195 @@
...
+        - 4
+        - 6
+    PLUGIN:
+        - foo
+        - bar
+# Will execute the same job 4 times with the following environment variables.
+# NODE_VERSION=4 PLUGIN=foo
+# NODE_VERSION=4 PLUGIN=bar
+# NODE_VERSION=6 PLUGIN=foo
+# NODE_VERSION=6 PLUGIN=bar
```
Comment (tkyi) [inline docs/reference/configuration.md]: Would move this comment above code to stay consistent with rest of examples

**Comment by stjohnjohnson** [comment]
All great comments, but to split out this work, I've reduced the pull-request to JUST an initial setup.

**Comment by d2lam** [comment]
That's a good idea!

**Comment by d2lam** [comment]
Sheep it! :+1
